src/api_keys/api.rs

```diff
@@ -43,21 +43,26 @@ pub async fn make_request(
     check_stale(user_make_request_info.timestamp, current_timestamp).await?;
 
     let api_key_info = app_state
-        .read_from_api_keys(&(request_author.0, user_make_request_info.api_url.clone()))?
+        .read_from_api_keys(&(
+            request_author.0,
+            user_make_request_info.api_url_base.clone(),
+        ))?
         .unwrap();
 
     let client = reqwest::Client::new();
+    let full_url = format!(
+        "{}{}",
+        user_make_request_info.api_url_base, user_make_request_info.api_url_extra
+    );
 
     let response = match user_make_request_info.http_verb.as_str() {
         "get" => {
-            let url = user_make_request_info
-                .api_url
-                .replace("xxxREPLACE_MExxx", &api_key_info);
+            let url = full_url.replace("xxxREPLACE_MExxx", &api_key_info);
```

Comment:
Not related to this PR but we could move the `.replace()` call to above the `match`, so we will allow giving API keys in the URL for all verbs.  Otherwise i cannot make a post request giving the api key in the url. Similarly i currently cannot make a get request giving the api key in the header.